myst_parser/docutils_renderer.py

```diff
@@ -421,25 +421,88 @@ def render_role(self, token):
             self.current_node += problematic
 
     def render_directive(self, token):
+        """parse fenced code blocks as directives.
+
+        Such a fenced code block starts with `{directive_name}`,
+        followed by arguments on the same line.
+
+        Directive options are read from a YAML block,
+        if the first content line starts with `---`, e.g.
+
+        ::
+
+            ```{directive_name} arguments
+            ---
+            option1: name
+            option2: |
+                Longer text block
+            ---
+            content...
+            ```
+
+        Or the option block will be parsed if the first content line starts with `:`,
+        as a YAML block consisting of every line that starts with a `:`, e.g.
+
+        ::
+
+            ```{directive_name} arguments
+            :option1: name
+            :option2: other
+
+            content...
+            ```
+
+        If the first line of a directive's content is blank, this will be stripped
+        from the content.
+        This is to allow for separation between the option block and content.
+
+        """
         name = token.language[1:-1]
         content = token.children[0].content
         options = {}
+        # get YAML options
         if content.startswith("---"):
             content = "\n".join(content.splitlines()[1:])
-            # get YAML options
             match = re.search(r"^-{3,}", content, re.MULTILINE)
             if match:
                 yaml_block = content[: match.start()]
-                content = content[match.end() :]  # TODO advance line number
+                content = content[match.end() + 1 :]  # TODO advance line number
             else:
                 yaml_block = content
                 content = ""
             try:
                 options = yaml.safe_load(yaml_block) or {}
-            except yaml.parser.ParserError:
-                # TODO handle/report yaml parse error
-                pass
-            # TODO check options are an un-nested dict?
+            except (yaml.parser.ParserError, yaml.scanner.ScannerError) as error:
+                msg_node = self.document.reporter.system_message(
+                    3, "Directive options:\n" + str(error), line=token.range[0]
+                )  # 3 is ERROR level
+                msg_node += nodes.literal_block(yaml_block, yaml_block)
+                self.current_node += [msg_node]
+                return
+            # TODO check options are an un-nested dict / json serialize ?
+        elif content.startswith(":"):
```

Comment:
Shall we lstrip here and in the while loop? I believe @jstac said he'd imagine indenting the : lines

Reply:
NOW you come in and review, after I moved the review restriction and went ahead and merged this PR 😆 Yeh this makes sense, especially in case the directives are nested, probably also need to handle indentations in the `---` YAML blocks.
If you could add an issue for it or make a new PR ta.

Reply:
HAH I knew you were gonna say that ;-)

I'll make it a PR

Reply:
FYI I’ve added #54, so a PR would close that